api/v1/monitor_types.go

```diff
@@ -22,6 +22,14 @@ import (
 
 // MonitorSpec defines the desired state of Tigera monitor.
 type MonitorSpec struct {
+
+	// PrometheusServiceListenPort defines the port that tigera-prometheus-service
+	// pod listens to and the TargetPort of calico-node-prometheus service.
+	// If not set it defaults to port 9090. In the scenario that the cluster
+	// is hosted on EKS and using Calico as it's CNI, tigera-prometheus-service
+	// will be HostNetwoked.  Configure this, if the default port value conflicts
+	// with any other ports currently used in the cluster's node network namespace.
+	PrometheusServiceListenPort int `json:"prometheusServiceListenPort,omitempty"`
```

Comment:
wondering if this is the right pattern to configure  the port since it will only configure an otherwise hidden  configuration (ie. serves connections through the calico-node-pometheus Service target port to the listen port of the Prometheus Proxy )  

another option is the configmap

 @tmjd or @caseydavenport  wdyt

Reply:
> In the scenario that the cluster is hosted on EKS and using Calico as it's CNI

I'd like to understand this more. Is this the only reason why we need the port to be configurable by the user?

The reason we make some things host-networked in this mode is because the EKS control-plane nodes attempt to send connections to those pods. Do the EKS control plane nodes initiate connections to the prometheus pod?

Reply:
>  Is this the only reason why we need the port to be configurable by the user?

yeah basically that is the major reason since prometheus proxy will be hostnetworked in that cluster configuration only, for the scenario that customer's might also have conflict with another hostnorked pod bind to this port

 I guess with exposing this configuration it can be configured at any time at the user's discretion  -> not sure if it's better to ignore this config if not hostnetwked

> Do the EKS control plane nodes initiate connections to the prometheus pod?
manager and voltron using the EKS managed kube apiserver as a proxy is the one connecting to the prometheus proxy pod

Reply:
ick. horrible.

Reply:
I don't see why we'd use a configmap for this - we have a CR available and it's valid user config. 

What's mostly sad is that we need this config option at all for a pretty niche scenario.

Reply:
I missed this conversation before. Was this before or after we had the call discussing the configuration of the port? @caseydavenport are you expecting this configuration to be in a ConfigMap or part of the Monitor resource?